### ISSUE TITLE
contacts Missing and *my worlds*   not populating 

### ISSUE THREAD

**Issue by peterthethinker** [Neos-Metaverse/NeosPublic#2598]
## Describe the bug
aftter the update today 

## Relevant issues
N/A

## To Reproduce
update?

## Expected behavior
Contacts should have my contacts in it 

## Log Files
DM me on discord to get them  as they do  have a  error! 
log  contain's PII

## Bug information (please complete the following information):
 - How often does it happen: [so far after today's update ]
 - Does the bug persist after restarting Neos? [Yes]
 - Neos Version: [version number]
 - Neos Platform: [Windows]

 - Regression: [Did this work before? Yes ( 4 days ago was fine)

## Additional context
[NA]

## Reporters:
[ peterthethinker ]

**Comment by ProbablePrime** [comment]
![](https://cdn.discordapp.com/attachments/439422161657135106/862118238301454346/unknown.png)

From a log file

**Comment by peterthethinker** [comment]
OK?   I dont do criptic  coder speek .

what is this

**Comment by H3BO3** [comment]
Prime is actually providing debug information for why this might be happening, so Froox can take a look.

You are not the only person experiencing this issue.

**Comment by ProbablePrime** [comment]
Oh just me providing some extra info i found in a log file from a user with the same problem.

**Comment by peterthethinker** [comment]
remember I had a stroke and I dont  think on all 12Cyl   so  I need context ;
)

**Comment by ProbablePrime** [comment]
No worries, Lots of people are experiencing this so thank you so much for opening a GH to track it.

**Comment by peterthethinker** [comment]
No  problem !

**Comment by Frooxius** [comment]
Thanks for the report, I've been looking into this, but unfortunately this is largely out of our control to fully fix at the moment. The Azure Cosmos Database we're using has been having a partial outage, which has been making a small portion of requests to fail, but enough to cause trouble like this.

They are working on resolution on their end, this is the current status as of now:
![image](https://user-images.githubusercontent.com/8838625/124734296-43384300-df15-11eb-9abd-0f6dc2e4c2b8.png)

Here is the request breakdown for past 24 hours from the database, there's a rise in the 500 errors (gray line):
![image](https://user-images.githubusercontent.com/8838625/124734671-a6c27080-df15-11eb-8868-0c678a9fb5e2.png)

I'm looking for workarounds in the meanwhile to work around these issues to perform a retry logic server-side and client-side in case of these issues and looking for some other mitigations, but it might take a bit.


When you or anyone else experience these problems, does it persist after restart?

**Comment by ProbablePrime** [comment]
Yes, it persisted after a few people I was talking to restarted.

**Comment by shiftyscales** [comment]
Yeah- I've also had a few friends note that it persisted after restart. When they were affected by it, they had some partial luck in doing things to refresh the contacts list like starting a search, then backing off of it, or searching someone up, and manually clicking on their name. Likely entirely coincidental/just tied to causing another get request to be ran.

**Comment by Frooxius** [comment]
A bit more info, here's service availability for last 24 hours, looks like it's going back up, so hopefully that'll resolve the issues: 
![image](https://user-images.githubusercontent.com/8838625/124738445-37e71680-df19-11eb-8754-103141347d57.png)

Still looking into workarounds and mitigations in case there are issues like this again.

**Comment by Frooxius** [comment]
Looks like it should be clearing up now and I haven't even made any changes yet! (got some tweaks upcoming though)

Database Response Errors:
![image](https://user-images.githubusercontent.com/8838625/124740822-6cf46880-df1b-11eb-974d-154b3ea47568.png)

This is the resulting server API errors, pretty much copies the graph:
![image](https://user-images.githubusercontent.com/8838625/124741001-944b3580-df1b-11eb-8c9e-2331d5438116.png)

Database availability: 
![image](https://user-images.githubusercontent.com/8838625/124741186-b9d83f00-df1b-11eb-9cb3-52ecc4805296.png)

**Comment by Frooxius** [comment]
Looks like we should be good now, the problem has been resolved from Azure's end without me even doing anything! :D

![image](https://user-images.githubusercontent.com/8838625/124743237-d2e1ef80-df1d-11eb-9fdf-f47c2c084d83.png)

Server errors back to 0: 
![image](https://user-images.githubusercontent.com/8838625/124743414-0290f780-df1e-11eb-8ad0-93eb0a7b9ad6.png)

Database:
![image](https://user-images.githubusercontent.com/8838625/124743652-39ffa400-df1e-11eb-91b4-45d64d8721fb.png)

I've also pushed some tweaks to increase some timeouts and handle some more errors like this, hopefully they'll help a bit in the future to deal with partial database outages like this more gracefully.

I'm closing this issue for now, but let me know if you still experience issues!
